packages/@aws-cdk/aws-logs/lib/query-definition.ts

```diff
@@ -58,62 +82,88 @@ export interface QueryStringProps {
 }
 
-interface QueryStringMap {
-  readonly fields?: string,
-  readonly parse?: string,
-  readonly filter?: string,
-  readonly stats?: string,
-  readonly sort?: string,
-  readonly limit?: Number,
-  readonly display?: string,
-}
-
 /**
  * Define a QueryString
  */
 export class QueryString {
   private readonly fields?: string[];
-  private readonly parse?: string;
-  private readonly filter?: string;
+  private readonly parse: string[];
+  private readonly filter: string[];
   private readonly stats?: string;
   private readonly sort?: string;
   private readonly limit?: Number;
   private readonly display?: string;
 
   constructor(props: QueryStringProps = {}) {
     this.fields = props.fields;
-    this.parse = props.parse;
-    this.filter = props.filter;
     this.stats = props.stats;
     this.sort = props.sort;
     this.limit = props.limit;
     this.display = props.display;
+
+    // Determine parsing by either the parseStatements or parse properties, or default to empty array
+    if (props.parseStatements) {
+      this.parse = props.parseStatements;
+    } else if (props.parse) {
+      this.parse = [props.parse];
+    } else {
+      this.parse = [];
+    }
+
+    // Determine filtering by either the filterStatements or filter properties, or default to empty array
+    if (props.filterStatements) {
+      this.filter = props.filterStatements;
+    } else if (props.filter) {
+      this.filter = [props.filter];
+    } else {
+      this.filter = [];
+    }
   }
 
   /**
   * String representation of this QueryString.
   */
   public toString(): string {
-    return noUndef({
-      fields: this.fields !== undefined ? this.fields.join(', ') : this.fields,
-      parse: this.parse,
-      filter: this.filter,
-      stats: this.stats,
-      sort: this.sort,
-      limit: this.limit,
-      display: this.display,
-    }).join('\n| ');
+    return [
+      this.buildQueryLine('fields', this.fields?.join(', ')),
+      ...this.buildQueryLines('parse', this.parse),
+      ...this.buildQueryLines('filter', this.filter),
+      this.buildQueryLine('stats', this.stats),
+      this.buildQueryLine('sort', this.sort),
+      this.buildQueryLine('limit', this.limit?.toString()),
+      this.buildQueryLine('display', this.display),
+    ].filter(
+      (queryLine) => queryLine !== undefined && queryLine.length > 0,
+    ).join('\n| ');
   }
-}
 
-function noUndef(x: QueryStringMap): string[] {
-  const ret: string[] = [];
-  for (const [key, value] of Object.entries(x)) {
-    if (value !== undefined) {
-      ret.push(`${key} ${value}`);
+  /**
+   * Build an array of query lines given a command and statement(s).
+   *
+   * @param command a query command
+   * @param statements one or more query statements for the specified command, or undefined
+   * @returns an array of the query string lines generated from the provided command and statements
+   */
+  buildQueryLines(command: string, statements?: string[]): string[] {
```

Comment:
My feeling is that since it's irrelevant outside of the QueryString class, it should be a method of that class. That being said, I missed the private prefix so can definitely add that in.